auth-foundation/src/test/java/com/okta/authfoundation/client/NetworkUtilsTest.kt

```diff
@@ -891,6 +903,45 @@ class NetworkingTest {
         assertThat(lastRetryEvent.response.peekBody(Long.MAX_VALUE).string()).isEqualTo("")
     }
 
+    @Test fun `prioritize custom okhttp interceptors when making requests`() = runBlocking {
+        var lastCalledInterceptor: String? = null
+
+        val customInterceptor = Interceptor {
+            lastCalledInterceptor = "customInterceptor"
+            it.proceed(it.request())
+        }
+
+        val okHttpClient = oktaRule.okHttpClient.newBuilder()
+            .addInterceptor(customInterceptor)
+            .build()
+
+        mockkObject(OidcUserAgentInterceptor)
+        val oidcRequestSlot = slot<Interceptor.Chain>()
+        every { OidcUserAgentInterceptor.intercept(capture(oidcRequestSlot)) } answers {
+            lastCalledInterceptor = "oidcInterceptor"
+            with(oidcRequestSlot.captured) {
+                proceed(request())
+            }
+        }
+
+        val oidcConfiguration = oktaRule.createConfiguration(okHttpClient)
+        oktaRule.enqueue(
+            path("/test"),
+        ) { response ->
+            response.setBody("response")
+        }
+        val request = Request.Builder()
+            .url(oktaRule.baseUrl.newBuilder().addPathSegments("test").build())
+            .build()
+
+        val credential = mockk<Credential>()
+        val oidcClient = OidcClient.create(oidcConfiguration, oktaRule.createEndpoints())
+            .withCredential(credential)
+
+        oidcClient.performRequest(String.serializer(), request) { /** Ignore response */ }
+        assertThat(lastCalledInterceptor).isEqualTo("customInterceptor")
```

Comment:
I confirmed that this tests fails without the changes made in OidcConfiguration

Reply:
TDD FTW! 🎉